runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -421,4 +426,54 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.showShortcutLevel())
+		{
+			return;
+		}
+
+		if (event.getTarget() == null || event.getTarget() == "")
+		{
+			return;
+		}
+
+		//Guarding against non-first option because agility shortcuts are always that type of event.
+		if (event.getType() != MenuAction.GAME_OBJECT_FIRST_OPTION.getId())
+		{
+			return;
+		}
+
+		MenuEntry[] menuEntries = client.getMenuEntries();
+
+		for (Obstacle nearbyObstacle : getObstacles().values())
+		{
+			for (int obId: nearbyObstacle.getShortcut().getObstacleIds())
+			{
+				if (obId == event.getIdentifier())
+				{
+					AgilityShortcut agilityShortcut = nearbyObstacle.getShortcut();
+					for (MenuEntry entry : menuEntries)
+					{
+						if (entry.getTarget().equalsIgnoreCase(event.getTarget()) && !entry.getTarget().toLowerCase().contains("(level-"))
```

Comment:
You can use the fact that "event" will always refer to the last menu option as the menu is built to avoid this for loop & unreliable string comparison.